hibernate-core/src/main/java/org/hibernate/internal/SessionFactoryImpl.java

```diff
@@ -247,8 +253,9 @@ public SessionFactoryImpl(final MetadataImplementor metadata, SessionFactoryOpti
 		for ( SessionFactoryObserver sessionFactoryObserver : options.getSessionFactoryObservers() ) {
 			this.observer.addObserver( sessionFactoryObserver );
 		}
-
-		this.typeResolver = metadata.getTypeResolver().scope( this );
+		TypeConfiguration typeConfiguration = metadata.getTypeConfiguration();
```

Comment:
IMO we should keep the TypeConfiguration either here or (more likely) on MetamodelImpl and access things from there.  E.g., here I would just drop the `typeResolver` field - we can always just get it from TypeConfiguration

Reply:
do you mean to add a `getTypeConfiguration() ` method to `MetamodelImplementor`?